### PR TITLE
Add NPR Rendering to GraphicsTools/StandardShader

### PR DESCRIPTION
**Pull request by HoloMoto**
## Overview

NPR functionality has been added to the light mode of GraphicsTools/Standard shaders.

Please see the following video for the actual operation.

https://youtu.be/rB6KaXwDAB4

The use of MeshOutline enables comic-like rendering.

　I would like to add new features to ToonShader in the future, but even at this point, we think it is possible to create some very interesting expressions.

　Personally, I think that the ability to use the existing StandardShader features in conjunction with the MeshOutline allows for a much greater variety of expressions!

## Changes
- Fixes: #108

## Verification

https://youtu.be/rB6KaXwDAB4

image

## Additional
 @Cameron-Micka 
　Sorry I'm a little late as I was learning the new LightMode process!

　TokyoHoloLensMeetup will be held on 10/12 in Japan!
I will be presenting MRGT in a 5 minute slot. I'm off to talk about what MRGT has to offer, including this NPR feature!

### CODE REVIEW

File: com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/GraphicsToolsLighting.hlsl
Comment (Cameron-Micka) [inline com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/GraphicsToolsLighting.hlsl:3]: Nitpick, looks like an 'E' was added.

```suggestion
// Unless expressly provided otherwise, the Software under this license is made available strictly on an AS IS BASIS WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED. Please review the license for details on these and other terms and conditions.

```

File: com.microsoft.mrtk.graphicstools.unity/Editor/ShaderGUIs/StandardShaderGUI.cs
Comment (Cameron-Micka) [inline com.microsoft.mrtk.graphicstools.unity/Editor/ShaderGUIs/StandardShaderGUI.cs:114]: I love that you put this option in a drop down so that the UI doesn't become cluttered with too many checkboxes. But there might be scenarios where people want to use [DistantLights](https://github.com/microsoft/MixedReality-GraphicsTools-Unity/pull/37) with Non-Photorealistic rendering?

If that is the case maybe we move this property down to a separate check box that gets displayed when "Lit - Directional" or "Lit - Distant" is selected. Just like how specular highlights and spherical harmonics are toggled: 
image

File: com.microsoft.mrtk.graphicstools.unity/Editor/ShaderGUIs/StandardShaderGUI.cs
Comment (Cameron-Micka) [inline com.microsoft.mrtk.graphicstools.unity/Editor/ShaderGUIs/StandardShaderGUI.cs:117]: nit: all caps to match the other keywords.

```suggestion
            public static readonly string lightModeNPR = "_NPR_RENDERING";

```

**Review by Cameron-Micka — COMMENTED** [review]
This is amazing! What a wonderful idea and great execution. I added a few comments mostly about how to charge how the feature is enabled in the shader GUI. Once that is addressed this looks good to merge. 

Also had fun doing some testing:
image

image

Also, if the Tokyo HoloLens Meetup was recored it would be fun to watch your segment.  😊

**Comment by HoloMoto** [comment]
@Cameron-Micka 
I've incorporated the Corrections !
It certainly needed to be used in conjunction with DistanceLight!
Also fixed the overlap between NPRRendering and Rendering. (I need to get better acquainted with the English language...)

File: com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/GraphicsToolsStandard.shader
Comment (Cameron-Micka) [inline com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/GraphicsToolsStandard.shader:37]: Could you also add this property to the `GraphicsToolsStandardCanvas` shader too? Without it you get this error in Unity:

> ArgumentException: Could not find MaterialProperty: '_NPR', Num properties: 117

File: com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/GraphicsToolsLighting.hlsl
Comment (Cameron-Micka) [inline com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/GraphicsToolsLighting.hlsl:20]: nitpick: You probably don't need this define?
```suggestion
#if defined(_DIRECTIONAL_LIGHT) || defined(_DISTANT_LIGHT)

```

**Review by Cameron-Micka — CHANGES_REQUESTED** [review]
Tested the latest changes, thank you for responding to the feedback. It looks wonderful! This is good to merge after fixing the "Could not find MaterialProperty" exception and fixing the small merge conflicts. 😊

**Comment by HoloMoto** [comment]
@Cameron-Micka 
Thank you!
I've corrected the two points you made!

**Review by Cameron-Micka — APPROVED** [review]
Looks great! This is ready to merge.

**Pull request merged**
